tests/pubsub/testscoring.nim

```diff
@@ -42,3 +38,441 @@ suite "GossipSub Scoring":
       gossipSub.gossipsub.peers(topic) == 0
       # also ensure we cleanup properly the peersInIP table
       gossipSub.peersInIP.len == 0
+
+  asyncTest "Time in mesh scoring (P1)":
+    const topic = "foobar"
+    var (gossipSub, conns, peers) =
+      setupGossipSubWithPeers(3, topic, populateMesh = true)
+    defer:
+      await teardownGossipSub(gossipSub, conns)
+
+    gossipSub.topicParams[topic] = TopicParams(
+      topicWeight: 1.0, # No multiplication factor
+      timeInMeshWeight: 1.0, # Weight = 1.0
+      timeInMeshQuantum: 1.seconds, # 1 second quantum
+      timeInMeshCap: 10.0, # Cap at 10.0
+    )
+
+    let now = Moment.now()
+
+    # Set different mesh times for peers
+    gossipSub.withPeerStats(peers[0].peerId) do(stats: var PeerStats):
+      stats.topicInfos[topic] = TopicInfo(
+        inMesh: true, graftTime: now - 2.seconds # 2 seconds in mesh
+      )
+
+    gossipSub.withPeerStats(peers[1].peerId) do(stats: var PeerStats):
+      stats.topicInfos[topic] = TopicInfo(
+        inMesh: true,
+        graftTime: now - 12.seconds, # 12 seconds in mesh (should be capped at 10)
```

Comment:
Mentioning the values in the comments can lend itself to the comment eventually not matching the values being used. Maybe instead could be something like  `now - (timeInMeshCap + 2.seconds) # (should be capped at timeInMesh`